### PR TITLE
(BOLT-757) Include custom facts and features in apply

### PR DESCRIPTION
**Pull request by MikaelSmith**
Ensure external facts are copied to targets during apply, that the apply
sets up Facter to find them, and that the lib dirs of modules are
available when resolving `puppet/feature` with the autoloader.

Also make it easier to debug `apply_catalog` by accepting a file name on
the CLI as an alternative to passing args via stdin.

### CODE REVIEW

File: libexec/apply_catalog.rb [hunk 1]
```diff
@@ -50,6 +50,13 @@
   end
 
   env = Puppet.lookup(:environments).get('production').override_with(modulepath: [moduledir])
+  # Needed to ensure features are loaded
+  env.each_plugin_directory do |dir|
+    $LOAD_PATH << dir unless $LOAD_PATH.include?(dir)
```
Comment (adreyer) [inline libexec/apply_catalog.rb]: This would have broken providers with non-relative requires too right?

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: Probably.

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: Extended the tests to include require from puppet_x.

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: But acceptance tests may not be working yet.

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: Now passes.

Comment (MikaelSmith) [reply libexec/apply_catalog.rb]: Tested with both Windows and Debian controllers.

**Pull request merged**